src/CreeDictionary/CreeDictionary/paradigm/test_manager.py

```diff
@@ -91,3 +153,17 @@ def bulk_lookup(self, analyses: Iterable[str]) -> dict[str, set[str]]:
     @staticmethod
     def lookup(analysis: str) -> str:
         return analysis
+
+
+def distinct_permutation(sequence):
+    """
+    Returns a permutation of the given sequence that is guaranteed to not be the same
+    order as the given input.
+    """
+    assert len(sequence) > 1, "there is no distinct permutation for this input"
+    original = list(sequence)
+    result = original.copy()
+
+    while result != original:
```

Comment:
Doesn't this guarantee that `result == original`?
```suggestion
    while result == original:
```

Reply:
yes 🤯  why... why does this code work...